src/main/java/com/deepoove/swagger/diff/compare/ParameterDiff.java

```diff
@@ -72,76 +65,49 @@ public ParameterDiff diff(List<Parameter> left,
           if (leftSchema instanceof RefModel && rightSchema instanceof RefModel) {
             String leftRef = ((RefModel) leftSchema).getSimpleRef();
             String rightRef = ((RefModel) rightSchema).getSimpleRef();
-            Model leftModel = oldDedinitions.get(leftRef);
-            Model rightModel = newDedinitions.get(rightRef);
+            Model leftModel = oldDefinitions.get(leftRef);
+            Model rightModel = newDefinitions.get(rightRef);
             String aRef = leftRef != null ? leftRef : rightRef;
-            ModelDiff diff = ModelDiff.buildWithDefinition(oldDedinitions, newDedinitions).diff(leftModel, rightModel, aRef);
-            changedParameter.setIncreased(diff.getIncreased());
-            changedParameter.setMissing(diff.getMissing());
-            changedParameter.setChanged(diff.getChanged());
+            ModelDiffResult modelDiff = ModelDiff.build(oldDefinitions, newDefinitions).diff(leftModel, rightModel, aRef);
+            changedParameter.setIncreased(modelDiff.getIncreased());
+            changedParameter.setMissing(modelDiff.getMissing());
+            changedParameter.setChanged(modelDiff.getChanged());
+            parameterDiffResult.setHasOnlyCosmeticChanges(modelDiff.hasOnlyCosmeticChanges());
```

Comment:
Will this overwrite the previously set values for every model diff?

Reply:
It wouldn't because a new `parameterDiffResult` object is made for every operation. This code snippet is only reached if a `BodyParameter` is a `RefModel`, and there can only be one `BodyParameter` per operation.